.sail.yml

```diff
@@ -13,7 +13,22 @@ tasks:
       - build
   test:
     image: "node:carbon"
-    command: 
+    command:
       - yarn
     args:
       - test
+  coverage:
+    image: "node:carbon"
+    command:
+      - yarn
+    args:
+      - coverage
+  codecov:
+    image: "node:carbon"
+    env:
+      - CODECOV_TOKEN
+    command:
+      - yarn
+    args:
+      - codecov
+      - --token=$(CODECOV_TOKEN)
```

Comment:
Isn’t this supposed to be ${} and not $() or is it some domain specific interpolation that’s not shell?

Reply:
Yep, that's originally what I thought too. Then I read Sail's docs on environment variables: https://sail.ci/docs/environment-variables. Check out the "Use Environment Variables in Task Properties" section, apparently they do interpolation using `$()`.

Reply:
Interesting! So, does that mean they're pre-processing the interpolation to be something like:

```sh
$ export CODECOV_TOKEN="hi"
$ yarn codecov --token=hi
```

?

Because if this was _bash_ with `$(STRING)` the `STRING` would be executed as a command. Curious if doing `- --token=${CODECOV_TOKEN}` produces

```sh
$ export CODECOV_TOKEN="hi"
$ yarn codecov --token=${CODECOV_TOKEN}
```

.. which seems like it should work too...

Reply:
My guess is they must be doing something like the above. I believe I ran an experiment where I had the CI run `echo ${CODECOV_TOKEN}` and it worked. The key to all of this is ensuring that the `env` key on the task is specified so that the environment variable you want to use is mounted: https://github.com/FormidableLabs/reason-urql/blob/v1/.sail.yml#L28-L29. Basically, the steps seem to be:

- Add env variables through Sail's app (app.sail.ci).
- Specify which variables to bring into scope in which task (using the `env` config option as shown above).
- Interpolate the variable using `$()`.

Sail does have a few other drawbacks from my research thus far.

- You can't see the build logs in their UI (app.sail.ci). The only way to see them is via the "Checks" tab on a PR. Which means you just don't get logs for non-PR'ed builds.
- Their environment variables UI seems badly broken. You can't actually save environment variables without opening the dev tools and bringing the Save button into view (or praying you tab navigate to the right out of view button 😬). 
![SailUI](https://user-images.githubusercontent.com/19421190/55675831-60d15a80-587d-11e9-8eab-17a6b8258d52.gif)
- Their deletion of environment variables doesn't seem to be working. If you delete variables in the UI and then refresh the page, your variables will still appear (and work in CI, meaning they're being persisted).
![SailNoDeletion](https://user-images.githubusercontent.com/19421190/55675854-e3f2b080-587d-11e9-98c5-cf91548d95ea.gif)

The CI itself seems to be running fine, but in the longer run I probably will advocate for us to migrate to Circle if these issues persist.